libr/bin/p/bin_elf.inc

```diff
@@ -677,27 +677,20 @@ static RBinReloc *reloc_convert(struct Elf_(r_bin_elf_obj_t) *bin, RBinElfReloc
 }
 
 static RList* relocs(RBinFile *bf) {
+	r_return_val_if_fail (bf && bf->o && bf->o->bin_obj, NULL);
 	RList *ret = NULL;
 	RBinReloc *ptr = NULL;
 	RBinElfReloc *relocs = NULL;
-	struct Elf_(r_bin_elf_obj_t) *bin = NULL;
-	ut64 got_addr;
-	int i;
-	if (!bf || !bf->o || !bf->o->bin_obj) {
-		return NULL;
-	}
-	bin = bf->o->bin_obj;
+	struct Elf_(r_bin_elf_obj_t) *bin = bf->o->bin_obj;
 	if (!(ret = r_list_newf (free))) {
 		return NULL;
 	}
 	/* FIXME: This is a _temporary_ fix/workaround to prevent a use-after-
 	 * free detected by ASan that would corrupt the relocation names */
 	r_list_free (imports (bf));
-	if ((got_addr = Elf_(r_bin_elf_get_section_addr) (bin, ".got")) == -1) {
+	ut64 got_addr = Elf_(r_bin_elf_get_section_addr) (bin, ".got");
+	if (got_addr == UT64_MAX) {
 		got_addr = Elf_(r_bin_elf_get_section_addr) (bin, ".got.plt");
-		if (got_addr == -1) {
```

Comment:
why?

Reply:
because it's checked below already

Reply:
if you mean that it is checked by https://github.com/radareorg/radare2/pull/17188/files#diff-be1d0d79bce9095b796c70fcfc154c1eL702 I'm not sure it works, because got_addr is ut64 and with your patch got_addr would remain UT64_MAX (-1). When the next `if` checks got_addr < 1 it would find it false.

On the other hand, on master when got_addr is UT64_MAX(-1) it gets replaced with 0, thus it enters the next if.

Reply:
Addr 0 is a valid address. -1 is not. Its green so it seems that is passing. I have more prs with tweaks in the elf parser to make it work better in such cases. And also making -1 become 0 is a bad idea.

Reply:
Why is it a bad idea? And why just removing this if should be better idea? I agree this code is not ideal, but I don't see a real reason for removing this `if` here and not, for example, few linew below at line 697. Also, removing this `if`, as I said in previous comment, changes the behaviour without an actual reason. I would reintroduce the `if`, unless you can improve the entire codebase of this function or prove this is better.

Reply:
Because 0 is a valid address and ut64-max not. which other reason do you need to keep the old behaviour? there's another PR introducing more changes to remove the assumption that 0 is not valid, but i wanted to add this change in here to confirm if it was breaking something in the suite. and seems like not, so when this gets merged i will continue the changes in the other pr

Reply:
I don't see why you have 3/4 elf-related PRs opened that are more or less working all on the same code and you keep referencing one from another. If all those changes are related, just do them in one PR. for example this line, I think, was also touched in another PR (where however you have also removed the other if and done some other things..)

Reply:
well, i have splitted the original PR into 5 different PRs. this is why i reference them and also why there are some mixed changes

Reply:
i can move this into the other ut64max pr if you want, i was trying to reduce the size of the original PR to get it into a shape that can be reviewed properly and i dont see whats wrong in this change here.

Reply:
please remove this change from this PR

Reply:
done